src/mapogcsld.cpp

```diff
@@ -4962,25 +4971,18 @@ static char *msSLDGetAttributeNameOrValue(const char *pszExpression,
     nLength = strlen(pszExpression);
     pszAttributeName = (char *)malloc(sizeof(char) * (nLength + 1));
     iValue = 0;
-    for (i = 0; i < nLength - 2; i++) {
-      if ((pszExpression[i] != szCompare[0] ||
-           pszExpression[i] != toupper(szCompare[0])) &&
-          (pszExpression[i] != szCompare2[0] ||
-           pszExpression[i] != toupper(szCompare2[0])))
-
-      {
-        pszAttributeName[iValue++] = pszExpression[i];
+    for (int i = 0; i < nLength; i++) {
+      // Check for comparison operator
+      if ((i + 1 < nLength && strncmp(&pszExpression[i], szCompare, 2) == 0) ||
+          (i + 1 < nLength && strncmp(&pszExpression[i], szCompare2, 2) == 0) ||
+          (i + 1 < nLength && strncmp(&pszExpression[i], szCompare3, 2) == 0)) {
```

Comment:
to preserve case insensitive comparisons:
```suggestion
      if ((i + 1 < nLength && EQUALN(&pszExpression[i], szCompare, 2)) ||
          (i + 1 < nLength && EQUALN(&pszExpression[i], szCompare2, 2)) ||
          (i + 1 < nLength && EQUALN(&pszExpression[i], szCompare3, 2))) {
```